sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -458,11 +472,23 @@ def collect_metrics(self):
                 # except Exception as e:
                 #     self.log.warning("Could not fetch metric %s : %s", metric.datadog_name, e)
 
-    def do_stored_procedure_check(self, proc):
+    def do_custom_queries(self):
+        """
+        Fetch metrics from custom SQL queries
+        """
+        tags = self.instance.get("tags", [])
+        for custom_query in self.custom_queries:
+
+            with self.connection.open_managed_default_connection():
```

Comment:
We might open many connections because of this. Could we keep a single connection throughout the whole collection of custom queries?